### PR TITLE
New home page design

### PR DESCRIPTION
**Pull request by ghost**
## Purpose
Better orienting the user in the key sections of the documentation by giving him quick access links and controls preview.

## Prototype
See the Figma prototype [here](https://www.figma.com/proto/ZqSOy1xVIoaZrAp6h6UZyu/Taipy-Doc-Lobby?scaling=min-zoom&page-id=0%3A1&node-id=0%3A1&starting-point-node-id=1%3A3)

### CODE REVIEW

**Review by jrobinAV — APPROVED** [review]

File: docs/index.md [hunk 1]
```diff
@@ -3,41 +3,266 @@ hide:
...
 ---
 
-# Home
+
+# Welcome to Taipy Documentation!
 
 Taipy is an innovative **low-code** package to create complete applications.
-It is composed of two main independent components:
-**Taipy GUI** and **Taipy Core**.
+This documentation is divided in three main sections:
```
Comment (FabienLelaquais) [inline docs/index.md:11]: Add a line break here.

File: docs/index.md [hunk 2]
```diff
@@ -3,41 +3,266 @@ hide:
...
+
+- [**"Getting Started"**](getting_started/index.md) provides a step-by-step opening to Taipy. Taipy features are leveraged as the application becomes more and more complex.
+- [**"User Manual"**](manuals/about.md) describes what the product concepts are, and how you can use them.
+- [**"Reference Manual"**](manuals/reference) gathers all the Python APIs that Taipy exposes.
```
Comment (FabienLelaquais) [inline docs/index.md]: Remove "Python" (there are REST and JavaScript APIs as well)

File: docs/index.md [hunk 3]
```diff
@@ -3,41 +3,266 @@ hide:
...
+These topics are the most visited ones, we thought you’d like to have a glance on them!
+
+<div class="tp-row tp-row--gutter-sm">
+  <div class="tp-col-12 tp-col-md-6 tp-col-lg-4 d-flex">
+    <a class="tp-content-card" href="getting_started/">
+      <img class="tp-content-card-icon icon-light" src="images/icons/flag-w.svg">
+      <img class="tp-content-card-icon icon-dark" src="images/icons/flag.svg">
+      <h3>Getting started</h3>
+      <p>
+        This tour shows you how to create an entire application using the two components of Taipy
```
Comment (FabienLelaquais) [inline docs/index.md]: "the two components"
->
"the two main components"

File: docs/index.md [hunk 4]
```diff
@@ -3,41 +3,266 @@ hide:
...
+      <span class="tp-content-card-readmore">Read more</span>
+    </a>
+  </div>
+  <div class="tp-col-12 tp-col-md-6 tp-col-lg-4 d-flex">
+    <a class="tp-content-card" href="manuals/reference">
+      <img class="tp-content-card-icon icon-light" src="images/icons/menu_book-w.svg">
+      <img class="tp-content-card-icon icon-dark" src="images/icons/menu_book.svg">
+      <h3>Reference Manual</h3>
+      <p>
+        Description of Taipy’s API.
```
Comment (FabienLelaquais) [inline docs/index.md]: "Description of Taipy’s API."
->
"Documentation for all APIs exposed in Taipy."

File: docs/index.md [hunk 5]
```diff
@@ -3,41 +3,266 @@ hide:
...
+      </p>
+      <span class="tp-content-card-readmore">Read more</span>
+    </a>
+  </div>
+  <div class="tp-col-12 tp-col-md-6 tp-col-lg-4 d-flex">
+    <a class="tp-content-card" href="manuals/gui/controls/">
+      <img class="tp-content-card-icon icon-light" src="images/icons/tune-w.svg">
+      <img class="tp-content-card-icon icon-dark" src="images/icons/tune.svg">
+      <h3>Controls</h3>
+      <p> Controls are UI objects for data vizualisation/interaction. </p>
```
Comment (FabienLelaquais) [inline docs/index.md]: "vizualisation/interaction" -> "vizualisation and interaction"

File: docs/index.md [hunk 6]
```diff
@@ -3,41 +3,266 @@ hide:
...
+      <p>
+        The Taipy GUI library provides Python classes to create powerful Web apps in minutes.
+      </p>
+      <span class="tp-content-card-readmore">Read more</span>
+    </a>
+  </div>
+</div>
 
-Taipy requires *Python 3.8* or newer.
+## Browse dozens of controls to build your UI!
```
Comment (FabienLelaquais) [inline docs/index.md:68]: This section looks pretty bad.,,
No ordering, no alignment, no visual hint for links...

I would certainly pick a few "good-to-have" controls (like charts or tables or menus) and leave a link to the visual elements home page to see them all.

File: docs/index.md [hunk 7]
```diff
@@ -3,41 +3,266 @@ hide:
...
 
-<p align="center">
-  <img alt="Taipy structure" src="images/taipy-structure.svg">
-</p>
+<ul class="tp-pills-list">
```
Comment (FabienLelaquais) [inline docs/index.md:70]: I love the hover effect here (tooltip)!
The big problem with the current implementation is that if we change the elements documentation, they cannot be updated in the doc. Which is bad.
Maybe these need to be generated after all.

File: docs/index.md [hunk 8]
```diff
@@ -3,41 +3,266 @@ hide:
...
+      <img alt="Taipy structure" src="images/taipy-structure-vector.svg" width="350">
+    </figure>
+  </div>
 
-Other packages offer additional functionality, such as a **REST API** that allows for deploying
-Taipy application on Web architectures.
+</div>
 
-The documentation for Taipy is divided in three main sections:
+Other packages offer additional functionality, such as a **REST API** that allows for deploying Taipy application on Web architectures.
```
Comment (FabienLelaquais) [inline docs/index.md]: "for deploying Taipy application on Web architectures"
->
"for deploying Taipy Core applications as a Web service"

File: docs/index.md [hunk 9]
```diff
@@ -3,41 +3,266 @@ hide:
...
 
-- [:material-arrow-right: **'Getting Started'**](getting_started/index.md)
-  provides a step-by-step opening to Taipy. Taipy features are leveraged as the application
-  becomes more and more complex.
-- [:material-arrow-right: **'User Manual'**](manuals/about.md)
-  describes what the product concepts are, and how you can use them.
-- [:material-arrow-right: **'Reference Manual'**](manuals/reference)
-  gathers all the Python APIs that Taipy exposes.
+Taipy requires *Python 3.8* or newer.
```
Comment (FabienLelaquais) [inline docs/index.md]: There was a time where Vincent wanted this information as soon as possible.
Please check with him showing the final doc home page.

See the 'About' page to see what I mean.

File: tools/assets/stylesheets/avaiga.css [hunk 10]
```diff
@@ -1,25 +1,40 @@
...
 }
+
 .md-header__button.md-logo img {
   width: 48px;
   height: 48px;
 }
 
-[data-md-color-scheme="slate"] {
-  --md-hue: 240;
+.md-content{
```
Comment (FabienLelaquais) [inline tools/assets/stylesheets/avaiga.css:21]: Please make the CSS code homogeneous: space before curly brace always or never.

File: tools/assets/stylesheets/avaiga.css [hunk 11]
```diff
@@ -1,25 +1,40 @@
...
+  overflow: initial;
 }
 
-a.ext-link:after {
+/* Display the right icon for the right theme */
+.icon-light{ display: none; }
```
Comment (FabienLelaquais) [inline tools/assets/stylesheets/avaiga.css]: Please make the CSS code homogeneous: block after selector on new line always or never.
Always would be pretty bad hu?

File: tools/assets/stylesheets/avaiga.css [hunk 12]
```diff
@@ -1,25 +1,40 @@
...
+
+a.ext-link::after {
   display: inline-block;
   content: url("../images/ext-link.svg");
   width: 0.6em;
   height: 0.6em;
   margin: 0 3px 0 3px;
 }
-[data-md-color-scheme="slate"] a.ext-link:after {
+[data-md-color-scheme="slate"] a.ext-link::after {
```
Comment (FabienLelaquais) [inline tools/assets/stylesheets/avaiga.css]: Just so that I learn something... why two colons?

File: tools/assets/stylesheets/avaiga.css [hunk 13]
```diff
@@ -32,6 +47,143 @@ a.ext-link:after {
...
+.tp-content-card{
+  flex: 0 0 100%;
+  position: relative;
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-start;
+  align-items: stretch;
+  padding: 1.25em;
+  border-radius: 0.5em;
+  background: var(--md-paper-bg-color);
```
Comment (FabienLelaquais) [inline tools/assets/stylesheets/avaiga.css:151]: May be a slight skew in the background color would make them look better.

**Review by FabienLelaquais — APPROVED** [review]
A few minor things... pretty neat globally.

File: tools/assets/stylesheets/avaiga.css (hunk 11 above)
Comment (WinzanaPL) [reply tools/assets/stylesheets/avaiga.css]: There's a middle ground here though: "only for one CSS rule déclaration".
It keeps the readability when multiple one-rule selectors are written after one another.
But let's go with never, then, if you prefer. I adapt. 😉 
Done.

File: tools/assets/stylesheets/avaiga.css (hunk 12 above)
Comment (WinzanaPL) [reply tools/assets/stylesheets/avaiga.css:37]: `::before` and `::after` are pseudo-elements (like `::placeholder`, for instance) that refer to a virtual CSS element whereas things like `:hover` or `:focus` are pseudo-classes that refer to an element's state.
`:after` may be well interpreted valid CSS, but the actual good practice for this is to make the distinction between pseudo-elements (two columns) and pseudo-classes (valid only with one column) in the code.

File: tools/assets/stylesheets/avaiga.css (hunk 13 above)
Comment (WinzanaPL) [reply tools/assets/stylesheets/avaiga.css:151]: Done

File: tools/assets/stylesheets/avaiga.css (hunk 10 above)
Comment (WinzanaPL) [reply tools/assets/stylesheets/avaiga.css:21]: Done.

File: docs/index.md (hunk 8 above)
Comment (WinzanaPL) [reply docs/index.md]: Done

File: docs/index.md (hunk 7 above)
Comment (WinzanaPL) [reply docs/index.md:68]: I'm not sure I understand the whole idea but I get the "must be generated" point. Let me know how you want to do that and if it needs my participation.

File: docs/index.md (hunk 2 above)
Comment (ghost) [reply docs/index.md]: Done

File: docs/index.md (hunk 1 above)
Comment (ghost) [reply docs/index.md:11]: Done

File: docs/index.md (hunk 4 above)
Comment (ghost) [reply docs/index.md]: Done

File: docs/index.md (hunk 3 above)
Comment (ghost) [reply docs/index.md]: Done

File: docs/index.md (hunk 5 above)
Comment (ghost) [reply docs/index.md]: Done

File: docs/index.md (hunk 9 above)
Comment (ghost) [reply docs/index.md]: It had actually been completely omitted in Astrid's mockup. I took upon myself to keep it here.
I put it back up right away.

File: docs/index.md (hunk 6 above)
Comment (ghost) [reply docs/index.md:68]: Astrid did that on the validated prototype. It didn't disturb me either visually. And Vincent validated that layout as well.
As for the alignment, elements are laid out on the span axis, so the fact they don't align vertically is not disturbing. It's like a text that wraps to a new line...
For the ordering, controls were not ordered in the prototype. I ordered them alphabetically during the integration.

File: docs/index.md (hunk 7 above)
Comment (ghost) [reply docs/index.md:70]: I'm not sure I understand the whole idea but I get the "must be generated" point. Let me know how you want to do that and if it needs my participation.

File: tools/assets/stylesheets/avaiga.css (hunk 12 above)
Comment (ghost) [reply tools/assets/stylesheets/avaiga.css]: `::before` and `::after` are pseudo-elements (like `::placeholder`, for instance) that refer to a virtual CSS element whereas things like `:hover` or `:focus` are pseudo-classes that refer to an element's state.
`:after` may be well interpreted valid CSS, but the actual good practice for this is to make the distinction between pseudo-elements (two colons) and pseudo-classes (valid only with one colon) in the code.

**Pull request merged**